src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java

```diff
@@ -217,6 +193,42 @@ public boolean equals(Object obj)
 		return isSimilar(bukkit) && this.amount == bukkit.getAmount();
 	}
 
+	@Override
+	public int hashCode()
+	{
+		if (type == ItemType.AIR)
+		{
+			return EMPTY.hashCode();
+		}
+		else
+		{
+			int hash = type.hashCode();
+			hash = hash * 31 + this.getAmount();
+			return hash;
```

Comment:
Also use durability and item meta to calculate the hash?

Reply:
I will investigate that, does sound reasonable to do, I do believe I tested on a real instance where I changed the item meta: This had no effect on item equality (which always should mean the hash needs to be the same)